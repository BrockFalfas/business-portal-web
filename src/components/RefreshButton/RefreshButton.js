import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'antd';

import TooltipButton from '~components/TooltipButton';
import './RefreshButton.scss';

export class RefreshButton extends React.Component {
  static propTypes = {
    handleRefresh: PropTypes.func.isRequired,
    isLoading: PropTypes.bool,
    type: PropTypes.string,
    size: PropTypes.string,
  };

  static defaultProps = {
    type: 'primary',
    isLoading: false,
    size: 'default',
  };
  render() {
    const { handleRefresh, isLoading, type, size } = this.props;

    return (
      <TooltipButton
        tooltip="Refresh page"
        type={type}
        size={size}
        onClick={handleRefresh}
        className="Refresh-button">
        <Icon type="sync" theme="outlined" spin={isLoading} />
      </TooltipButton>
    );
  }
}

export default RefreshButton;
