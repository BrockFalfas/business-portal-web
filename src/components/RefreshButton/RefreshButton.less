.Refresh-button {
  margin-left: 0.6rem;
}
