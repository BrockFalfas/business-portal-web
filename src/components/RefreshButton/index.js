import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Button } from 'antd';

import './RefreshButton.css';

class RefreshButton extends React.Component {
  static propTypes = {
    handleRefresh: PropTypes.func.isRequired,
    isLoading: PropTypes.bool,
    type: PropTypes.string,
    size: PropTypes.string,
  };

  static defaultProps = {
    type: 'primary',
    isLoading: false,
    size: 'default',
  };
  render() {
    const { handleRefresh, isLoading, type, size } = this.props;

    return (
      <Button type={type} size={size} onClick={handleRefresh} className="Refresh-button">
        <Icon type="sync" theme="outlined" spin={isLoading} />
      </Button>
    );
  }
}

export default RefreshButton;
