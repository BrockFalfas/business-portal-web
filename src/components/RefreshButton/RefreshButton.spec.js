import React from 'react';
import { shallow } from 'enzyme';

import { RefreshButton } from './RefreshButton';

const initRefreshButton = overrides => {
  const mockProps = {
    handleRefresh: jest.fn(),
  };
  const wrapper = shallow(<RefreshButton {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('component: <RefreshButton/>', () => {
  describe('RefreshButton', () => {
    it('should render without crashing', () => {
      const { wrapper } = initRefreshButton();
      expect(wrapper).toMatchSnapshot();
    });
    it('should find button', () => {
      const { wrapper } = initRefreshButton();
      expect(wrapper.find('Button')).toBeTruthy();
    });
    it('should find button', () => {
      const { wrapper, mockProps } = initRefreshButton();
      wrapper.find('Button').simulate('click');
      expect(mockProps.handleRefresh).toHaveBeenCalled();
    });
  });
});
