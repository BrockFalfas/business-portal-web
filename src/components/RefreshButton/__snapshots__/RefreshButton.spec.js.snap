// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <RefreshButton/> RefreshButton should render without crashing 1`] = `
<Button
  className="Refresh-button"
  onClick={[Function]}
  size="default"
  type="primary"
>
  <Icon
    spin={false}
    theme="outlined"
    type="sync"
  />
</Button>
`;
